prov/verbs/configure.m4

```diff
@@ -19,7 +19,28 @@ AC_DEFUN([FI_VERBS_CONFIGURE],[
 				[],
 				[$verbs_PREFIX],
 				[$verbs_LIBDIR],
-				[verbs_ibverbs_happy=1],
+				[
+				AC_MSG_CHECKING(if libibverbs is linkable by libtool)
+				file=conftemp.$$.c
+				rm -f $file conftemp
+				cat > $file <<EOF
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
```

Comment:
Yes, that comment is from configure scripted generated for libfabric. I think it has this comment for all AC_CHECK_LIB macro expansions. I think the comment partially applies here since we are overriding the ibv_open_device from libibverbs. I have a couple of questions.
At first, including <infiniband/verbs.h> in the code snippet threw an error:
```
conftemp.4225.c:8:6: error: conflicting types for 'ibv_open_device'
 char ibv_open_device ();
      ^
In file included from conftemp.4225.c:7:0:
/home/arun/include/infiniband/verbs.h:1033:21: note: previous declaration of 'ibv_open_device' was here
 struct ibv_context *ibv_open_device(struct ibv_device *device);
                     ^
```
That's when I looked into the generated configure script for libfabric and found that it doesn't have any includes. So with the include removed, can we be sure that the link test is correct? To address your question on the comment below, I haven't tested this for the scenario in issue #2070. I guess, testing that would answer my question too.